language/tools/move-cli/src/main.rs

```diff
@@ -382,20 +418,40 @@ fn run(
     let log_context = NoContextLog::new();
     let mut session = vm.new_session(&state);
 
-    let res = session.execute_script(
-        script_bytes,
-        vm_type_args.clone(),
-        vm_args,
-        signer_addresses.clone(),
-        &mut cost_strategy,
-        &log_context,
-    );
+    let script_type_parameters = vec![];
+    let script_parameters = vec![];
+    let res = if let Some(script_name) = script_name_opt {
```

Comment:
if-let-else :(

Reply:
> > This sounds very convenient, but enforcing the uniqueness sounds potentially hard (e.g., I recompile a script file to rename a script to foo, but some other script file I didn't recompile already contains a script named foo).
> 
> You would basically be calling `move_compile` on `src/`, reading over the compiled_units and bailing if there is more than one script with that name. That flow might be a tad easier than eagerly enforcing uniqueness
> 
> > This is what the PR adds, minus the .move part + forces you to say --script-name . I'm not sure if you can configure structopt to let you drop the --script-name part unless every invocation of run requires it.
> 
> I think we are still slightly off here. I would not allow `--script-name` to point to a module's script fun. This would only be used for normal scripts.
> 
> Script functions would only be invoked via `module run <addr>::<module>::<name>`

I think it makes sense to support both `move run <addr>::<module>::<name>` and `move run <path_to_mv> --script-name <fun_name>`. 
- One useful workflow would be `move view /path/to/mv`, see a script fun `foo` you want to call, then `move run /path/to/mv --script-name foo`. I think that becomes less convenient/intuitive if you are forced to type `<addr>::<module>::<name>`
- The existing mode of running `move run` is to point at a script file. It seems logical to allow that same mode for (or a close approximation) for modules.